File: frontend/src/types/task.ts
export type DayOfWeek =
  | "Monday"
  | "Tuesday"
  | "Wednesday"
  | "Thursday"
  | "Friday"
  | "Saturday"
  | "Sunday";

export type Task = {
  id: string;
  title: string;
  description: string | null;
  day_of_week: DayOfWeek | null;
  status: "todo" | "done";
  source_voice_note_id: string | null;
  created_at: string;
  updated_at: string;
};

export type TaskUpdatePayload = {
  status?: "todo" | "done";
};

export type VoiceNote = {
  id: string;
  original_filename: string;
  transcription_text: string;
  created_at: string;
};

export type UploadVoiceNoteResponse = {
  transcription_text: string;
  voice_note: VoiceNote;
  extracted_tasks: Task[];
};

File: backend/app/schemas/task.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

DayOfWeek = Literal[
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
    "Sunday",
]

TaskStatus = Literal["todo", "done"]


class TaskCreate(BaseModel):
    title: str
    description: str | None = None
    day_of_week: DayOfWeek | None = None
    due_date: date | None = None
    status: TaskStatus = "todo"
    source_voice_note_id: str | None = None


class TaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    day_of_week: DayOfWeek | None = None
    due_date: date | None = None
    status: TaskStatus | None = None


class TaskRead(TaskCreate):
    id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
